app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailActivity.java

```diff
@@ -51,4 +62,30 @@ public boolean onOptionsItemSelected(MenuItem item) {
     public AndroidInjector<Fragment> supportFragmentInjector() {
         return dispatchingAndroidInjector;
     }
+
+    @Override
+    public void onBackPressed() {
+        Fragment infoFragment = fragmentManager.findFragmentByTag(INFO_FRAGMENT_TAG);
+        if (infoFragment != null) {
+            if (saveAlertDialog == null) {
+                saveAlertDialog = new AlertDialog.Builder(new ContextThemeWrapper(this, R.style.AlertDialog))
+                    .setMessage(getString(R.string.save_changes))
+                    .setPositiveButton(getString(R.string.save), (dialog, which) -> {
+                        UpdateOrganizerInfoViewModel viewModel = ViewModelProviders.of(infoFragment, viewModelFactory)
+                            .get(UpdateOrganizerInfoViewModel.class);
+                        viewModel.updateOrganizer();
+                        dialog.dismiss();
+                        super.onBackPressed();
```

Comment:
Why call the super method here?

Reply:
Because the fragment needs to be closed for both cases, or we can explicitly call `#popBackStack`.